third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -3266,23 +3266,48 @@ impl<'c, 'l, 'v, C: CustomSerializer> serde::Serialize
     for SerializationReadyValue<'c, 'l, 'v, MoveStructLayout, Vec<ValueImpl>, C>
 {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let values = &self.value;
-        let fields = self.layout.fields();
-        if fields.len() != values.len() {
-            return Err(invariant_violation::<S>(format!(
-                "cannot serialize struct value {:?} as {:?} -- number of fields mismatch",
-                self.value, self.layout
-            )));
-        }
-        let mut t = serializer.serialize_tuple(values.len())?;
-        for (field_layout, value) in fields.iter().zip(values.iter()) {
+        let values = self.value.as_slice();
```

Comment:
Can you add serialization-deserialization roundtrip tests? We did something similar here https://github.com/aptos-labs/aptos-core/blob/a8c816fa068c85574337a6479c210c2602c19836/third_party/move/move-vm/types/src/delayed_values/delayed_field_id.rs#L208 for testing replacing fields with identifiers for delayed values, would be nice to see enums also tested - at least a few to_bcs, from_bcs calls.